osu.Game/Online/Multiplayer/MultiplayerClient.cs

```diff
@@ -413,7 +413,7 @@ async Task IMultiplayerClient.UserJoined(MultiplayerRoomUser user)
 
                 UserJoined?.Invoke(user);
                 RoomUpdated?.Invoke();
-            });
+            }, false);
```

Comment:
This is a required change since otherwise user joins could be misordered with `UserStateChanged` (which also specifies this `false` parameter). This immediately fails some tests that call both methods in quick order, but consider this a bug fix in general.